output/cloud/expv2/hdr.go

```diff
@@ -129,7 +129,12 @@ func histogramAsProto(h *histogram, time int64) *pbcloud.TrendHdrValue {
 		// if the current and the previous indexes are not consecutive
 		// consider as closed the current on-going span and start a new one.
 		if diff := indexes[i] - indexes[i-1]; diff > 1 {
-			spans = append(spans, &pbcloud.BucketSpan{Offset: diff, Length: 1})
+			spans = append(spans, &pbcloud.BucketSpan{
+				// Offset counts the empty buckets between two non-zero buckets.
+				Offset: diff - 1,
```

Comment:
nitPick: I would actually renaame `diff` to `zerosBetween` and then check it against 0, but keep the single line `append`